### ISSUE TITLE
Initialization module gave 'nan' for normal-light images

### ISSUE THREAD

**Issue by kazukochan** [AndersonYong/URetinex-Net#18]
When I tried to train an initialization module for normal-light images from LOL datasets, the forward computation result of decom.py turned out to be nan(both R and L). Input images have been normalized to [0,1], so I am very confused. Everything is fine with training for low-light images from LOL datasets. Is it a gradient explosion? Have you got any idea what is going on there?

**Comment by kazukochan** [comment]
I found the reason, this phenomenon is caused by outdated numpy, simply updating numpy by pip install --upgrade numpy solves the issue~

**Comment by kazukochan** [comment]
Seems that the structure-aware smooth constraint loss produces nan during back propogation. With this loss term, network parameters become nan after loss.backward() and optimizer.step(). Perhaps my computation of the gradient of the illumination component is wrong

**Comment by kazukochan** [comment]
I found the reason by using torch.autograd.set_detect_anomaly(True). I used torch.sqrt() when computing image gradient, stop using it solves the issue.

**Comment by kazukochan** [comment]
I guess what happens here is that a zero is divided because the derivative of a sqrt has got the sqrt on the denominator.
